riscv/tests/common/mod.rs

```diff
@@ -37,30 +38,26 @@ pub fn verify_riscv_asm_string<T: FieldElement, S: serde::Serialize + Send + Syn
 
     test_plonky3_pipeline::<T>(pipeline.clone());
 
-    /* Commenting out these tests for now because auto witgen does not handle
-    * partial witnesses well with Poseidon.
-        // verify using executor generated witness
-        // TODO remove the guard once the executor is implemented for BB
-        if T::known_field().unwrap() == KnownField::GoldilocksField {
-            let analyzed = pipeline.compute_analyzed_asm().unwrap().clone();
-            let pil = pipeline.compute_optimized_pil().unwrap();
-            let fixed = pipeline.compute_fixed_cols().unwrap().clone();
-            let execution = powdr_riscv_executor::execute(
-                &analyzed,
-                &pil,
-                fixed,
-                Default::default(),
-                pipeline.data_callback().unwrap(),
-                &[],
-                None,
-                None,
-            );
-            pipeline.rollback_from_witness();
-            let executor_trace: Vec<_> = execution.trace.into_iter().collect();
-            let pipeline = pipeline.add_external_witness_values(executor_trace);
-            test_plonky3_pipeline::<T>(pipeline);
-        }
-    */
+    // verify executor generated witness
+    if executor_witgen {
```

Comment:
This new parameter was included so we can test the riscv witgen in the supported cases only (it won't work if unimplemented coprocessors poseidon2/keccak/arith are enabled, because they mess with the memory machine witness)

Reply:
it also only works for GL